api/opencl_architecture.asciidoc

```diff
@@ -199,10 +199,12 @@ A command submitted to a device will not launch until prerequisites that
 constrain the order of commands have been resolved.
 These prerequisites have three sources:
 
-  * They may arise from commands submitted to a command-queue that constrain
-    the order in which commands are launched.
-    For example, commands that follow a command-queue barrier will not
-    launch until all commands prior to the barrier are complete.
+  * The first source of prerequisites is implicit dependencies between commands
+    enqueued to the same command-queue which arise as follows:
+  ** Commands enqueued after a command-queue barrier have all commands enqueued
+     before the barrier as prerequisites.
+  ** Commands enqueued in an in-order command-queue have all commands enqueued before
+     them as prerequisites.
```

Comment:
I'm fine with this as-is and I think this is correct, but I've always found in-order queues easier to think of using ideas from  mathematical induction, and simpliy thinking about an implicit dependency on the previous command.  Logically this means that each command has an implicit dependency on all previous commands also, but if you draw this as a picture (really, a DAG) then it's the difference between one dependence arrow between each command (node in the graph) vs. N dependence arrows between commands (one connecting to each previous graph node).

I'd at least consider rephrasing to something like:

> * Commands in an in-order queue have an implicit dependency on the completion of the preceding command as a prerequisite.

Reply:
I like the idea. This could come in handy for an extension I have on the back burner. I've updated the description but used a different wording.